dartagnan/src/main/java/com/dat3m/dartagnan/Dartagnan.java

```diff
@@ -243,8 +243,9 @@ public static String generateResultSummary(VerificationTask task, ProverEnvironm
                 }
                 if (props.contains(LIVENESS) && FALSE.equals(model.evaluate(LIVENESS.getSMTVariable(encCtx)))) {
                     summary.append("============ Liveness violation found ============\n");
-                    for(Event e : p.getEvents(Label.class)) {
-                        if(e.is(Tag.SPINLOOP) && TRUE.equals(model.evaluate(encCtx.execution(e)))) {
+                    for(Event e : p.getEvents(CondJump.class)) {
+                        if(e.is(Tag.SPINLOOP) && TRUE.equals(model.evaluate(encCtx.execution(e)))
+                            && TRUE.equals(model.evaluate(encCtx.jumpCondition((CondJump)e)))) {
```

Comment:
You can directly do `for (CondJump e : p.getEvents(CondJump.class))` (the method returns a correctly typed list of events), so you don't need to do the cast anymore.